server/data/data-ldap/src/test/java/org/apache/james/user/ldap/ReadOnlyUsersLDAPRepositoryTest.java

```diff
@@ -46,183 +52,633 @@
     static final Username UNKNOWN = Username.of("unknown");
     static final String BAD_PASSWORD = "badpassword";
 
-    DomainList domainList;
+    @Nested
+    class WhenEnableVirtualHosting implements AbstractUsersRepositoryContract.WithVirtualHostingContract {
```

Comment:
Maybe split the conttract to distinguish RO/RW tests

Reply:
Sorry what do you mean RO/RW ?

Reply:
He is doing the same suggestion than I did, in a more summarized form.

https://github.com/linagora/james-project/pull/3174#discussion_r388147704

Reply:
Read-Only and Read-Write

Reply:
okay, Thank, I just figured it out :)